src/utils/createAdaptorShape.js

```diff
@@ -0,0 +1,7 @@
+export default function createAdaptorShape(PropTypes) {
+  return PropTypes.shape({
+    resolve: PropTypes.func.isRequired,
```

Comment:
:+1: this is a well thought out simplification. I had originally wanted to do this, but thought it would be too jarring of a change.